ports/esp32/mphalport.h

```diff
@@ -36,10 +36,15 @@
 #include "freertos/task.h"
 
 // The core that the MicroPython task(s) are pinned to.
-// Until we move to IDF 4.2+, we need NimBLE on core 0, and for synchronisation
-// with the ringbuffer and scheduler MP needs to be on the same core.
-// See https://github.com/micropython/micropython/issues/5489
+// Now that we have IDF 4.2.1+, we are once again able to pin to core 1
+// and avoid the Wifi/BLE timing problems on the same core.
+// Best effort here to remain backwards compatible in rare version edge cases...
+// See https://github.com/micropython/micropython/issues/5489 for history
+#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(4, 2, 1)
```

Comment:
Will it work with IDF 4.2?

Reply:
Hey @dpgeorge good call out...I just tried with 4.2.1 because it was the latest...but I wlll try 4.2 tonight and report back.  If it works in my use case I will update this PR.

Reply:
@dpgeorge I can confirm that my problem is also resolved with this patch and IDF 4.2.0, so I have update the PR to match.  Thanks!